From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Ruikai (Rick) Liu" <rickliu2000@outlook.com>
Date: Wed, 15 Sep 2021 22:20:35 +0200
Subject: [ARCHEOLOGY] Add NanoPi R2C support (#3138)

> X-Git-Archeology: > recovered message: > * Add NanoPi R2C support
> X-Git-Archeology: > recovered message: > * Bring NanoPi R2C, R2S to edge
> X-Git-Archeology: - Revision 687c3639183e23e06406739af1684fdeb3efa454: https://github.com/armbian/build/commit/687c3639183e23e06406739af1684fdeb3efa454
> X-Git-Archeology:   Date: Wed, 15 Sep 2021 22:20:35 +0200
> X-Git-Archeology:   From: Ruikai (Rick) Liu <rickliu2000@outlook.com>
> X-Git-Archeology:   Subject: Add NanoPi R2C support (#3138)
> X-Git-Archeology:
> X-Git-Archeology: - Revision dd51f9f2afcbc83a3e10b32eb6a5061d91d1558e: https://github.com/armbian/build/commit/dd51f9f2afcbc83a3e10b32eb6a5061d91d1558e
> X-Git-Archeology:   Date: Tue, 09 Nov 2021 18:06:34 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bump imx6, xu4, rockchip64 and jetson-nano to 5.15 (#3238)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ac8fc4385594d59257ee9dffd9efa85e3497fa7d: https://github.com/armbian/build/commit/ac8fc4385594d59257ee9dffd9efa85e3497fa7d
> X-Git-Archeology:   Date: Sat, 26 Feb 2022 07:46:44 +0100
> X-Git-Archeology:   From: Piotr Szczepanik <piter75@gmail.com>
> X-Git-Archeology:   Subject: Switch rockchip64 current to linux 5.15.y (#3489)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 897674aa74bce0326ed7fe06f5336bf4709a8a1f: https://github.com/armbian/build/commit/897674aa74bce0326ed7fe06f5336bf4709a8a1f
> X-Git-Archeology:   Date: Tue, 03 May 2022 08:27:32 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bump and freeze kernel at last known working versions (#3736)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 597d2dac11f00d9070a4e49d6bad1b2244e36cb3: https://github.com/armbian/build/commit/597d2dac11f00d9070a4e49d6bad1b2244e36cb3
> X-Git-Archeology:   Date: Sat, 28 May 2022 07:56:22 +0200
> X-Git-Archeology:   From: Jianfeng Liu <liujianfeng1994@gmail.com>
> X-Git-Archeology:   Subject: update rockchip64-edge to 5.18 (#3814)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 8c6641e7b79f0d50acdc306d140e586a4e923cf0: https://github.com/armbian/build/commit/8c6641e7b79f0d50acdc306d140e586a4e923cf0
> X-Git-Archeology:   Date: Wed, 03 Aug 2022 22:22:55 +0200
> X-Git-Archeology:   From: Jianfeng Liu <liujianfeng1994@gmail.com>
> X-Git-Archeology:   Subject: update rockchip64 edge to 5.19 (#4039)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 6765f734cc4a22aeaa9f99a3ad28c8c322de26f6: https://github.com/armbian/build/commit/6765f734cc4a22aeaa9f99a3ad28c8c322de26f6
> X-Git-Archeology:   Date: Tue, 25 Oct 2022 11:26:51 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bump rockchip64 edge to 6.0.y (#4337)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 92f1a22d76b987afa7ba555d5b509adc51d689e7: https://github.com/armbian/build/commit/92f1a22d76b987afa7ba555d5b509adc51d689e7
> X-Git-Archeology:   Date: Fri, 16 Dec 2022 13:38:13 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Re-add rockchip64 6.0 patches (#4575)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 34ae84fac5d0b66a1ab2d1e51534b7beb13ef245: https://github.com/armbian/build/commit/34ae84fac5d0b66a1ab2d1e51534b7beb13ef245
> X-Git-Archeology:   Date: Fri, 05 May 2023 14:22:00 +0200
> X-Git-Archeology:   From: amazingfate <liujianfeng1994@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip64 edge to v6.3
> X-Git-Archeology:
---
 arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-rev06.dts | 89 ++++++++++
 1 file changed, 89 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-rev06.dts b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-rev06.dts
new file mode 100644
index 000000000000..21be5a2c5030
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-rev06.dts
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 FriendlyElec Computer Tech. Co., Ltd.
+ * (http://www.friendlyarm.com)
+ */
+
+/dts-v1/;
+
+#include "rk3328-nanopi-r2-rev00.dts"
+
+/ {
+	model = "FriendlyElec NanoPi R2C";
+	compatible = "friendlyelec,nanopi-r2", "rockchip,rk3328";
+};
+
+&mach {
+	hwrev = <6>;
+	model = "NanoPi R2C";
+};
+
+&rgmiim1_pins {
+	rockchip,pins =
+		/* mac_txclk */
+		<1 RK_PB4 2 &pcfg_pull_none_8ma>,
+		/* mac_rxclk */
+		<1 RK_PB5 2 &pcfg_pull_none>,
+		/* mac_mdio */
+		<1 RK_PC3 2 &pcfg_pull_none_2ma>,
+		/* mac_txen */
+		<1 RK_PD1 2 &pcfg_pull_none_8ma>,
+		/* mac_clk */
+		<1 RK_PC5 2 &pcfg_pull_none_2ma>,
+		/* mac_rxdv */
+		<1 RK_PC6 2 &pcfg_pull_none>,
+		/* mac_mdc */
+		<1 RK_PC7 2 &pcfg_pull_none_2ma>,
+		/* mac_rxd1 */
+		<1 RK_PB2 2 &pcfg_pull_none>,
+		/* mac_rxd0 */
+		<1 RK_PB3 2 &pcfg_pull_none>,
+		/* mac_txd1 */
+		<1 RK_PB0 2 &pcfg_pull_none_8ma>,
+		/* mac_txd0 */
+		<1 RK_PB1 2 &pcfg_pull_none_8ma>,
+		/* mac_rxd3 */
+		<1 RK_PB6 2 &pcfg_pull_none>,
+		/* mac_rxd2 */
+		<1 RK_PB7 2 &pcfg_pull_none>,
+		/* mac_txd3 */
+		<1 RK_PC0 2 &pcfg_pull_none_8ma>,
+		/* mac_txd2 */
+		<1 RK_PC1 2 &pcfg_pull_none_8ma>,
+
+		/* mac_txclk */
+		<0 RK_PB0 1 &pcfg_pull_none>,
+		/* mac_txen */
+		<0 RK_PB4 1 &pcfg_pull_none>,
+		/* mac_clk */
+		<0 RK_PD0 1 &pcfg_pull_none>,
+		/* mac_txd1 */
+		<0 RK_PC0 1 &pcfg_pull_none>,
+		/* mac_txd0 */
+		<0 RK_PC1 1 &pcfg_pull_none>,
+		/* mac_txd3 */
+		<0 RK_PC7 1 &pcfg_pull_none>,
+		/* mac_txd2 */
+		<0 RK_PC6 1 &pcfg_pull_none>;
+};
+
+/delete-node/ &rtl8211e;
+
+&gmac2io {
+	phy-handle = <&ethphy3>;
+       snps,reset-delays-us = <0 15000 50000>;
+       tx_delay = <0x22>;
+	rx_delay = <0x12>;
+	mdio {
+
+		ethphy3: ethernet-phy@3 {
+			compatible = "ethernet-phy-id0000.011a",
+				     "ethernet-phy-ieee802.3-c22";
+			reg = <3>;
+                       interrupt-parent = <&gpio2>;
+			interrupts = <RK_PC4 IRQ_TYPE_EDGE_FALLING>;
+			//reset-gpios = <&gpio1 RK_PC2 GPIO_ACTIVE_LOW>;
+			keep-clkout-on;
+		};
+	};
+};
-- 
Armbian

